### PR TITLE
Add Flowdock plugin

### PR DESCRIPTION
**Pull request by mikecrittenden**
This adds a plugin for auto-posting to Flowdock. It's similar to the Slack plugin.

### CODE REVIEW

**Comment by mikecrittenden** [comment]
The test failures don't seem to be legit.

**Comment by matthutchinson** [comment]
Nice work, i'll set myself up with a flowdock account and try it out.

**Comment by mikecrittenden** [comment]
👍 Thanks! I've used it on a few repos with a few different Flowdock rooms so far, all successfully.

File: lib/lolcommits/plugins/lol_flowdock.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      print "Open the URL below and issue a token for your user (Personal API token):\n"
+      print "https://flowdock.com/account/tokens\n"
+      print "Enter the generated token below, then press enter: \n"
+      code = STDIN.gets.to_s.strip
+      print "Enter the machine name of the flow you want to post to from this repo.\n"
+      print "Go to https://www.flowdock.com/account and click Flows, then click the flow, then get the machine name from the URL:\n"
+      flow = STDIN.gets.to_s.strip
+      print "Enter the name of the organization for this Flowdock account.\n"
+      organization = STDIN.gets.to_s.strip
+
```
Comment (matthutchinson) [inline lib/lolcommits/plugins/lol_flowdock.rb]: Along with strip, I think you also need to `downcase` the flow and organisation captured above.  

When I tried with my test new flowdock account, I mistakenly configured with capitalised letters and the endpoint returned 403 (so it is case sensitive).

**Comment by matthutchinson** [comment]
All looks OK, but there was one issue I had when testing (see the other comment).

**Comment by mikecrittenden** [comment]
You rock. Thanks for not only testing but debugging when it didn't work. I'll fix this up tomorrow.

**Comment by mikecrittenden** [comment]
@matthutchinson downcase has been added to flow and organization.

**Comment by matthutchinson** [comment]
Thanks, looks good, will merge and get a new release out soon, would like to also include PR #315

**Comment by matthutchinson** [comment]
Now released with v[0.8.1](https://github.com/mroth/lolcommits/blob/master/CHANGELOG.md#081-11-october-2016) - sorry it took so long to get out

**Comment by matthutchinson** [comment]
Hi @mikecrittenden 👋

So you may (or may not) have noticed that we [recently extracted](https://github.com/mroth/lolcommits/issues?utf8=✓&q=label%3Aarchitecture+label%3Aplugin+) all lolcommit plugins to external gems!

🏅 Since you contributed this flowdock plugin, I've added you (as an admin) to the new [lolcommits-flowdock](https://github.com/lolcommits/lolcommits-flowdock) repo.

Of course **you are under no obligation** to maintain this code or gem going forward. This is just a quick message to explain what has happened and give you the opportunity to take ownership again.

If you're interested in picking this up, **send me your email address** (to matt/at/hiddeloop.com) and I'll add you as a new gem owner (with publishing rights) [here](https://rubygems.org/gems/lolcommits-flowdock).

Otherwise I will continue to maintain the gem, and make sure things are kept up to date.

---

To get an idea of how gem plugins work, check out the `sample_plugin` [README](https://github.com/lolcommits/lolcommits-sample_plugin) / [RubyDocs](http://www.rubydoc.info/github/lolcommits/lolcommits-sample_plugin) or browse through [other plugin](https://github.com/lolcommits) code.

The main reasons for extraction were to:

 * remove the ever-growing gem dependency list in the main lolcommits gem
 * start to unify configuration code for plugins (common helpers, approaches to OAuth etc.)
 * make lolcommit plugin development easier
 * add more comprehensive plugin tests

 That's it, any questions just let me know.

 Happy coding!

 Matt

**Pull request merged**